pkg/sql/sem/builtins/builtins.go

```diff
@@ -321,6 +321,53 @@ var builtins = map[string]builtinDefinition{
 				"Supports encodings 'UTF8' and 'LATIN1'.",
 		}),
 
+	// https://www.postgresql.org/docs/9.0/functions-binarystring.html#FUNCTIONS-BINARYSTRING-OTHER
+	"get_bit": makeBuiltin(tree.FunctionProperties{Category: categoryString},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"val", types.VarBit}, {"position", types.Int}},
+			ReturnType: tree.FixedReturnType(types.Int),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				val := tree.MustBeDBitArray(args[0])
+				position := tree.MustBeDInt(args[1])
+
+				// Checking whether index asked is inside array of bits.
+				if position < 0 || uint(position) >= val.BitLen() {
+					return nil, pgerror.Newf(
+						pgcode.ArraySubscript, "bit index %d out of valid range (0..%d)", position, val.BitLen()-1)
+				}
+				return tree.NewDInt(tree.DInt(val.BitArray.String()[position] - '0')), nil
+			},
+			Info: "extract a particular bit from a given array of bits",
+		}),
+
+	// https://www.postgresql.org/docs/9.0/functions-binarystring.html#FUNCTIONS-BINARYSTRING-OTHER
+	"set_bit": makeBuiltin(tree.FunctionProperties{Category: categoryString},
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"val", types.VarBit},
+				{"position", types.Int},
+				{"toSet", types.Int},
+			},
+			ReturnType: tree.FixedReturnType(types.VarBit),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				val := tree.MustBeDBitArray(args[0])
+				position := tree.MustBeDInt(args[1])
+				toSet := tree.MustBeDInt(args[2])
+
+				// Checking whether index asked is inside array of bits.
+				if position < 0 || uint(position) >= val.BitLen() {
+					return nil, pgerror.Newf(
+						pgcode.ArraySubscript, "bit index %d out of valid range (0..%d)", position, val.BitLen()-1)
+				}
+				// Converted BitArray to rune format, and update the given index.
+				encodedString := []rune(val.BitArray.String())
+				encodedString[position] = rune('0' + toSet)
```

Comment:
For this operation and the above operation, I think it would be worth it to implement a get and set operation on a `bitarray.BitArray` rather than doing these various casts to strings and back.

Reply:
Sure, 
I was thinking of doing the same because doing it by casting quite was unnatural.
along with this, I had broken change into 2 commits one for adding methods in ```bitarray.BitArray``` and another for adding ```builtin functions```

Reply:
hii @rohany, 
my changes are not able to pass TeamCity build failed, I had tried to debug but can not able to narrow down the cause, log report generated says 
``` Process exited with code 2 (Step: Verify source tarball (Command Line))
  Step Verify source tarball (Command Line) failed
Step 4/4: Assert Workspace Clean (Command Line)````
and 
```  Process exited with code 1 (Step: Assert Workspace Clean (Command Line))
  Step Assert Workspace Clean (Command Line) failed```
can you point me where to log for debugging 

thanks

Reply:
I see this error in the output:
```
[Test archive] fatal: Not a git repository (or any of the parent directories): .git
[14:38:05]	[Test archive] github.com/cockroachdb/cockroach/pkg/util/bitarray
[14:38:05]	[Test archive] # github.com/cockroachdb/cockroach/pkg/util/bitarray
[14:38:05]	[Test archive] pkg/util/bitarray/bitarray.go:552:35: invalid operation: uint64(1) << (numBitsPerWord - 1 - index % numBitsPerWord) (shift count type int, must be unsigned integer)
[14:38:05]	[Test archive] pkg/util/bitarray/bitarray.go:570:38: invalid operation: uint64(1) << (numBitsPerWord - 1 - index % numBitsPerWord) (shift count type int, must be unsigned integer)
[14:38:05]	[Test archive] pkg/util/bitarray/bitarray.go:572:49: invalid operation: uint64(toSet) << (numBitsPerWord - 1 - index % numBitsPerWord) (shift count type int, must be unsigned integer)
[14:38:05]	[Test archive] make[1]: *** No rule to make target 'bin/execgen', needed by 'pkg/col/coldata/vec.eg.go'.  Stop.
[14:38:05]	[Test archive] make[1]: *** Waiting for unfinished jobs....
```